sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -231,7 +231,7 @@ case class NewInstance(
       s"new $className($argString)"
     }
 
-    if (propagateNull) {
+    if (propagateNull && argGen.nonEmpty) {
```

Comment:
a small fix: we should not go into this branch if `argGen` is empty, or the `argsNonNull` will be `!()`, which is wrong.